### PR TITLE
Hotfix timestamps snr

### LINKED ISSUE #427: Bug in `_parse_timestamps_and_detectors` prevents use of dict timestamps
When trying to create detector states from a dictionary of timestamps the following happens:
```
  File "pyfstat/snr.py", line 436, in _parse_timestamps_and_detectors
    detectors = list(timestamps.key())
AttributeError: 'dict' object has no attribute 'key'
```

- [x] Check why that wasn't covered by the test suite.
- [x] Change `key` to `keys`.

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Closes #427 

Add minimal tests for `DetectorStates` to make sure that
1. Different inputs actually work.
2. Different inputs are consistent amongst each other.

### CODE REVIEW

File: pyfstat/snr.py [hunk 1]
```diff
@@ -56,21 +56,20 @@ class SignalToNoiseRatio:
...
-                )
-        elif self.assumeSqrtSX is not None:
+        if have_noise_weights == have_sqrtSX:
             raise ValueError(
-                "Need either `assumeSqrtSX` or `noise_weights` to account for background noise"
+                "Need either `assumeSqrtSX` or `noise_weights` to account for background noise."
             )
 
+        self.Tsft = self.detector_states.data[0].deltaT
+        if not have_noise_weights:
```
Comment (dbkeitel) [inline pyfstat/snr.py:68]:
```suggestion
        if have_sqrtSX:
```

Just to be in principle ready for a non-binary future.

File: pyfstat/snr.py [hunk 2]
```diff
@@ -430,10 +429,19 @@ def _parse_timestamps_and_detectors(self, timestamps, Tsft, detectors):
...
 
             if detectors is not None:
-                raise ValueError("`timestamps`' keys are redundant with `detectors`. ")
+                raise ValueError("`timestamps`' keys are redundant with `detectors`.")
+            for ifo in timestamps:
+                try:
+                    lalpulsar.FindCWDetector(name=ifo, exactMatch=True)
+                except Exception:
+                    raise ValueError(
+                        "Invalid detector name in timestamps. "
```
Comment (dbkeitel) [inline pyfstat/snr.py:438]:
```suggestion
                        f"Invalid detector name {ifo} in timestamps. "
```

**Review by dbkeitel — CHANGES_REQUESTED** [review]
Two small amendments. Otherwise looks good.

Also for the future I'd suggest to first commit the new test so that one can see the breakage in the logs, then the fix after. (On lalsuite I was told the opposite is preferred; but as we're a smaller project, the break-first approach seems more informative than a hassle.)

File: pyfstat/snr.py (hunk 1 above)
Comment (Rodrigo-Tenorio) [reply pyfstat/snr.py:68]: I agree. There is a XOR right before this, so `not have_noise_weights` <-> `have_sqrtSX`.

**Review by dbkeitel — APPROVED** [review]

**Pull request merged**